shared/pages/Swap/DepositWindow/DepositWindow.js

```diff
@@ -31,48 +33,79 @@ export default class DepositWindow extends Component {
 
     this.state = {
       swap,
-      checking: false,
-      missingBalance: 0,
-      isPressCtrl: false,
+      remainingBalance: 0,
       flow: swap.flow.state,
+      checkingPayment: false,
       isAddressCopied: false,
       isBalanceFetching: false,
       address: currencyData.address,
-      currency: currencyData.currency,
       scriptAddress: flow.scriptAddress,
       scriptBalance: flow.scriptBalance,
+      balance: flow.balance,
       currencyFullName: currencyData.fullName,
       sellAmount: (this.swap.sellAmount.toNumber() + 0.00005),
-      unconfContractBalance: flow.scriptUnconfirmedBalance || 0,
-      falseBalance: currencyData.balanc || flow.scriptBalance || 0,
     }
   }
 
   componentDidMount() {
-    const { sellAmount, scriptBalance } = this.state
+    const { sellAmount, scriptBalance, balance } = this.state
 
     let checker
 
+    const availableBalance = swap.sellCurrency === 'BTC' ? scriptBalance : balance
+
     checker = setInterval(() => {
-      if (scriptBalance <= sellAmount) {
+      if (availableBalance <= sellAmount) {
         this.updateBalance()
       } else {
         clearInterval(checker)
       }
     }, 5000)
   }
 
-  updateBalance = () => {
+  updateBalance = async () => {
     const { swap } =  this.props
-    const { sellAmount, scriptBalance, currency, flowBalance, unconfBalance } =  this.state
+    const { sellAmount, scriptBalance, flowBalance, unconfBalance, address, scriptAddress } =  this.state
+
+    const coinsWithDynamicFee = [
+      'eth',
+      'ltc',
+      'btc',
+    ]
+
+    const curBalance = await actions[swap.sellCurrency.toLowerCase()].getBalance()
+    const tokenBalance = await actions.token.getBalance(swap.sellCurrency.toLowerCase())
+    const balance = this.props.tokenItems.map(item => item.currency).includes(swap.sellCurrency) ? tokenBalance : curBalance
+
+    if (coinsWithDynamicFee.includes(swap.sellCurrency.toLowerCase())) {
+
+      const minAmount = await helpers[swap.sellCurrency.toLowerCase()].estimateFeeValue({ method: 'swap', speed: 'normal' })
 
-    const missingBalance = (sellAmount - scriptBalance).toFixed(6)
+      const requiredAmount = Number(sellAmount) + Number(minAmount) > 0 ?  Number(sellAmount) + Number(minAmount) : 0
```

Comment:
```suggestion
      const requiredAmount = sellAmount.plus(minAmount).isGreaterThan(0) ?  sellAmount.plus(minAmount) : BigNumber(0)
```

BigNumber нужно превращать в Number только перед выводом! Никакие операции нельзя производить с Number!

Reply:
Лучше anyAmount хранить в string, так как Number округляет с погрешностью. String можно закинуть в BN для расчетов, а выводить можно и строку. #numberненужен